reframe/core/schedulers/slurm.py

```diff
@@ -366,3 +381,8 @@ def _extract_attribute(self, attr_name, node_descr):
         else:
             raise JobError("could not extract attribute '%s' from "
                            "node description" % attr_name)
+
+    def __str__(self):
+        return ('Slurm node {{name: {0}, partitions: {1}, '
+                'active features: {2}}}').format(
+            self._name, self._partitions, self._active_features)
```

Comment:
I would simply return the node name here, because this would should up nicely and naturally in '%s'.
If you want debug information, you could simply implement the `__repr__` and use the utility we provide from the framework that does print all the internal fields:

```python
import reframe.core.debug as debug

def __repr__(self):
    return debug.repr(self)
```